### PR TITLE
7pm-3 search messages between two dates

### PR DESCRIPTION
**Pull request by khalidmihlar**
For this pull request, we created a new page in the Nav Bar titled Search By Date in Analyze Slack Data Category Drop Down Menu. From here, we implemented two inputs for both a start date and an end date, with the ability to click on a calender to input dates or manually type the dates. Once both dates are set and the Search Button is clicked on the frontend webpage, two UNIX timestamps are sent to the backend to filter through message and access all messages sent between those two dates. From here, the backend sends all those messages back to the front end in chronological order, where it would be displayed neatly on a table for the viewers experience.

### CODE REVIEW

**Comment by andrewhlu** [comment]
Just an FYI - if you need to make changes to a PR, please do NOT close and remake your pull request. Instead, just push new changes to your branch as normal - it'll automatically update the PR.
This helps us keep the PR list nice and clean!

**Comment by andrewhlu** [comment]
Previous PR: #282 
@pconrad previously requested changes:
> Could you make a more human readable title? (thisisnotagoodexampleofahumanreadbletitle)
> 
> Also: In the PR description, don't assume that the reader already knows the story. Start the story at the beginning not in the middle. Tell the reader, in a few words, what the PR is about.
> 
> Example:
> 
> In this PR, we add a new page, accessible from the ___ option on the ___menu, that allows the user to specify a start and end date, and see ______

File: javascript/src/main/pages/AnalyzeMessageData/SearchMessagesByUser.js [hunk 1]
```diff
@@ -23,6 +23,7 @@ const SearchMessagesByUser = () => {
         fetchWithToken(url, getToken, options)
             .then((response) => {
                 setSearchResults(response);
+                console.log(response);
```
Comment (mdeirossi) [inline javascript/src/main/pages/AnalyzeMessageData/SearchMessagesByUser.js:26]: I think all of the console logs should be removed

File: javascript/src/main/pages/AnalyzeMessageData/SearchBetweenDates.js [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+    const handleSearchDateOnSubmit = (event) => {
+        event.preventDefault();
+        const url = `/api/members/messages/datesearch?searchDate=${searchDate}&searchDate2=${searchDate2}`;
+        const options = {
+            method: 'GET',
+        }
+        fetchWithToken(url, getToken, options)
+            .then((response) => {
+                setSearchResults(response);
+                console.log(response);
```
Comment (mdeirossi) [inline javascript/src/main/pages/AnalyzeMessageData/SearchBetweenDates.js:29]: Here's another one

File: javascript/src/main/pages/AnalyzeMessageData/SearchBetweenDates.js [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+            })
+    };
+    console.log(typeof searchDate);
+    console.log(searchDate);
```
Comment (mdeirossi) [inline javascript/src/main/pages/AnalyzeMessageData/SearchBetweenDates.js:33]: Two more

**Review by mdeirossi — CHANGES_REQUESTED** [review]
We should probably remove the console logs used in testing

File: javascript/src/main/App.js [hunk 4]
```diff
@@ -52,6 +53,8 @@ function App() {
           <AuthorizedRoute path="/member/analyzemessages/countmessages" exact component={CountMessagesByUser} authorizedRoles={["admin", "member"]} />
           <AuthorizedRoute path="/member/analyzemessages/messagehistogram" exact component={HistogramOfMessagesByUser} authorizedRoles={["admin", "member"]} />
           <AuthorizedRoute path="/member/analyzemessages/searchmessages" exact component={SearchMessagesByUser} authorizedRoles={["admin", "member"]} />
+          <AuthorizedRoute path="/member/analyzemessages/searchmessages" exact component={SearchMessagesByUser} authorizedRoles={["admin", "member"]} />
```
Comment (TranNicholas) [inline javascript/src/main/App.js:56]: is this copy pasted by accident?

Comment (alvin-tu) [reply javascript/src/main/App.js:56]: That was part of the original code, there were multiple duplicate lines. I may have deleted it and pushed it, but when we pulled from main it added that line back

Comment (TranNicholas) [reply javascript/src/main/App.js:56]: Ok, can you remove it, in main there is only one line, so maybe git got confused and thought you made a new line so it added a duplicate. https://github.com/ucsb-cs156-s21/proj-mapache-search/blob/main/javascript/src/main/App.js#L54

**Review by TranNicholas — DISMISSED** [review]
LGTM

**Review by mdeirossi — DISMISSED** [review]
LGTM

**Review by brianlee4725 — APPROVED** [review]
LGTM

File: javascript/src/main/pages/AnalyzeMessageData/SearchBetweenDates.js [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+    const handleSearchDateOnChange = (event) => {
+        setSearchDate((new Date(event.target.value).getTime() / 1000).toString());
+    };
+
+    const handleSearchDateOnChange2 = (event) => {
+        setSearchDate2((new Date(event.target.value).getTime() / 1000).toString());
+    };
+    const handleSearchDateOnSubmit = (event) => {
+        event.preventDefault();
+        const url = `/api/members/messages/datesearch?searchDate=${searchDate}&searchDate2=${searchDate2}`;
```
Comment (andrewhlu) [inline javascript/src/main/pages/AnalyzeMessageData/SearchBetweenDates.js:22]: I don't really like the names `searchDate` and `searchDate2` - it's kinda unclear what these are at first glance. I think `startDate` and `endDate` would be better names - I can see them being used as the form labels, so there isn't any reason that I'm aware of that prevents us from using these variable names in our code. Can we make this change everywhere?

**Review by andrewhlu — CHANGES_REQUESTED** [review]

**Review by mdeirossi — DISMISSED** [review]
LGTM

**Review by mdeirossi — APPROVED** [review]
LGTM

**Review by pconrad — APPROVED** [review]
LGTM: tested on QA

**Pull request merged**